src/ray/gcs/gcs_server/gcs_actor_distribution.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/common/task/scheduling_resources.h"
+#include "ray/common/task/task_spec.h"
+#include "ray/gcs/gcs_server/gcs_actor_manager.h"
+#include "ray/gcs/gcs_server/gcs_actor_scheduler.h"
+#include "ray/gcs/gcs_server/gcs_node_manager.h"
+#include "ray/gcs/gcs_server/gcs_resource_manager.h"
+#include "ray/gcs/gcs_server/gcs_resource_scheduler.h"
+#include "ray/gcs/gcs_server/gcs_table_storage.h"
+#include "src/ray/protobuf/gcs.pb.h"
+
+namespace ray {
+namespace gcs {
+
+/// `GcsActorWorkerAssignment` represents the assignment from one or multiple actors to a
+/// worker process.
+/// TODO(Chong-Li): It contains multiple slots, and each of them can bind to an actor.
+class GcsActorWorkerAssignment
+    : public std::enable_shared_from_this<GcsActorWorkerAssignment> {
+ public:
+  /// Construct a GcsActorWorkerAssignment.
+  ///
+  /// \param node_id ID of node on which this gcs actor worker assignment is allocated.
+  /// \param acquired_resources Resources owned by this gcs actor worker assignment.
+  /// \param is_shared A flag to represent that whether the worker process can be shared.
+  GcsActorWorkerAssignment(const NodeID &node_id, const ResourceSet &acquired_resources,
+                           bool is_shared);
+
+  const NodeID &GetNodeID() const;
+
+  const ResourceSet &GetResources() const;
+
+  bool IsShared() const;
+
+ private:
+  /// ID of node on which this actor worker assignment is allocated.
+  NodeID node_id_;
```

Comment:
could these fields be const qualified?

Reply:
For the time being, they are const. I'll do the change.